src/opentracing/sampler/CompositeSampler.ts

```diff
@@ -0,0 +1,34 @@
+import Sampler from './Sampler';
+
+class CompositeSampler implements Sampler<any> {
+    private samplers: Array<Sampler<any>>;
+    private operator: SamplerCompositionOperator;
+
+    constructor(samplers: Array<Sampler<any>>, operator: SamplerCompositionOperator) {
+        this.samplers =  samplers;
+        this.operator = operator === undefined ? SamplerCompositionOperator.OR : operator;
+    }
+
+    isSampled(arg?: any): boolean {
+        if (this.operator === SamplerCompositionOperator.AND) {
+            let isSampled: boolean = true;
+            for (const sampler of this.samplers) {
+                isSampled = isSampled && sampler.isSampled(arg);
+            }
+            return isSampled;
+        } else {
+            let isSampled: boolean = true;
+            for (const sampler of this.samplers) {
+                isSampled = isSampled || sampler.isSampled(arg);
```

Comment:
`sampler.isSampled(arg)` should be at the left side of `||` operator. Otherwise if `sampled` becomes `true`, right side is not evaluated. And this can be problematic for some samplers like `CountAwareSampler` as it should always be executed to increase the counter.